src/meta/api/src/schema_api_impl.rs

```diff
@@ -1075,6 +1077,58 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         })
     }
 
+    #[tracing::instrument(level = "debug", ret, err, skip_all)]
+    async fn update_index(&self, req: UpdateIndexReq) -> Result<UpdateIndexReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+        loop {
+            trials.next().unwrap()?;
+
+            let index_id_key = IndexId {
+                index_id: req.index_id,
+            };
+
+            let (index_meta_seq, _) = get_index_meta_by_id_or_err(self, &index_id_key).await?;
+
+            if index_meta_seq == 0 {
+                return Err(KVAppError::AppError(AppError::UnknownIndex(
+                    UnknownIndex::new(&req.index_name, "update_index"),
+                )));
+            }
+
+            // Update index meta: update (index_id) -> index_meta
+            let condition = vec![txn_cond_seq(&index_id_key, Eq, index_meta_seq)];
+
+            let if_then = vec![txn_op_put(
+                &index_id_key,
+                serialize_struct(&req.index_meta)?,
+            )];
+
+            let txn_req = TxnRequest {
+                condition,
+                if_then,
+                else_then: vec![],
+            };
```

Comment:
It looks like it does need a transaction, a single `upsert-kv` would be good enough, IMHO